bin/github-review.sh

```diff
@@ -45,53 +41,130 @@ if ! [[ $1 =~ ^[0-9]+$ ]]; then
+                fi
+                # Check the pattern for the set of three lines
+                if [ "$pattern_correct" = false ]; then
+                    echo "Unknown lines added or removed" 1>&2
+                    exit 1
+                fi
+                # Reset the pattern_correct flag for the next set of lines
+                pattern_correct=true
+                ;;
+        esac
+        # Increment the counter
+        ((counter++))
+    done
+fi
+
 echo "Verifying Deployment Asset"
-gh pr diff $pr --patch | git apply --include 'src/assets/**'
+gh pr diff $pr --patch | git apply --include 'src/assets/v'$version'/**'
 
 # Getting default addresses, address on the chain and checking code hash.
+deploymentTypes=($(jq -r --arg c "$chainid" '.networkAddresses[$c]' "$versionFiles"))
+if [[ $deploymentTypes == "[" ]]; then
+    deploymentTypes=($(jq -r --arg c "$chainid" '.networkAddresses[$c][]' "$versionFiles"))
+fi
 for file in "${versionFiles[@]}"; do
-    defaultAddress=$(jq -r '.defaultAddress' "$file")
-    networkAddress=$(jq -r --arg c "$chainid" '.networkAddresses[$c]' "$file")
-    if [[ $defaultAddress != $networkAddress ]]; then
-        echo "ERROR: "$file" default address is not the same as the one created for the chain id" 1>&2
-        exit 1
-    fi
-    defaultCodeHash=$(jq -r '.codeHash' "$file")
-    networkCodeHash=$(cast keccak $(cast code $networkAddress --rpc-url $rpc))
-    if [[ $defaultCodeHash != $networkCodeHash ]]; then
-        echo "ERROR: "$file" code hash is not the same as the one created for the chain id" 1>&2
-        exit 1
-    fi
+    for deploymentType in "${deploymentTypes[@]}"; do
+        defaultAddress=$(jq -r --arg t "$deploymentType" '.deployments[$t].address' "$file")
+        defaultCodeHash=$(jq -r --arg t "$deploymentType" '.deployments[$t].codeHash' "$file")
+        networkCodeHash=$(cast code "$defaultAddress" --rpc-url "$rpc" | tr -d '\n' | cast keccak)
+        if [[ $defaultCodeHash != $networkCodeHash ]]; then
+            echo "ERROR: "$file" ("$defaultAddress") code hash ("$defaultCodeHash") is not the same as the one created for the chain id ("$networkCodeHash")" 1>&2
+            exit 1
+        fi
+    done
 done
 
 echo "Network addresses & Code hashes are correct"
 
 git restore --ignore-unmerged -- src/assets
 
 # NOTE/TODO
-# - We should still manually verify there is no extra code in the PR.
-# - We could fetch the version, chain id and rpc from the PR (needs a standard format and possibly a tag in PR).
-# - We can approve PR using Github CLI. Should only be added after all manual tasks can be automated.
-# - Supporting zkSync and alternative deployment addresses for 1.3.0 contracts.
-# - If there are changes in any other path other than src/assets, the script should show an error.
-# - We should ensure there are not more than a single chain ID being added in a PR.
+# - We should still manually verify there is no removal of deployment types for a single chain.
```

Comment:
I'm not sure I understand (it late on a Friday).

Reply:
Currently, if there are two versions mentioned for a chain, and a malicious person deletes one deployment type from all files. It still will pass the current review script. Thus, as a next step of improvement, we should be able to detect if a chain is removed or not from the existing list (only applicable to `v1.3.1` for now).